src/t/d4.clj

```diff
@@ -0,0 +1,164 @@
+(ns t.d4
+  (:gen-class)
+  (:require
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]))
+
+(def sample
+"MMMSXXMASM
+MSAMXMSMSA
+AMXSXMAAMM
+MSAMASMSMX
+XMASAMXAMM
+XXAMMXXAMA
+SMSMSASXSS
+SAXAMASAAA
+MAMMMXMMMM
+MXMXAXMASX")
+
+(defn parse-input
+  [input]
+  (->> input
+       cljstr/split-lines
+       (map-indexed (fn [i l]
+                      (->> l
+                      (map-indexed (fn [ii letter] {letter [[i ii]]}))
+                      (apply merge-with into))))
+       (apply merge-with into)))
```

Comment:
This one is probably very subjective, but it took me a bit of time to figure out what `into` is doing here. Perhaps:
```suggestion
                      (apply merge-with (fnil conj [])))))
       (apply merge-with (fnil conj []))))
```
Though that requires changing the inner function to `{letter [i ii]}` instead.

I think overall I'd have written this as:

```clojure
(let [lines (cljstr/split-lines input)]
  (->> (for [y (range (count input))
            :let [line (nth lines y)] 
            x (range (count line))
            :let [c (nth line x)]]
         [c [y x]])
       (reduce (fn [acc [c p]] (update acc c (fnil conj #{}) p)))))
```
(untested), but I don't really have any concrete argument for whether that's better.

Reply:
this is because of the way I reach that point step by step, and the `(apply merge-with into)` is in the examples for `merge-with` so I used it, modified the input so it worked and moved on